### ISSUE TITLE
RDS: When ManageMasterUserPassword is set to true, MasterUserPassword must not be specified.

### ISSUE THREAD

**Issue by rhbecker** [aws-cloudformation/cfn-lint#2562]
### CloudFormation Lint Version

0.72.9

### What operating system are you using?

Windows

### Describe the bug

With `ManageMasterUserPassword` set to `true`, the linter incorrectly reports that ...

> [cfn-lint] E2521: Property MasterUserPassword should exist with MasterUsername

The [documentation for ManageMasterUserPassword](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html#cfn-rds-dbcluster-managemasteruserpassword) specifies a constraint ...

> Can't manage the master user password with AWS Secrets Manager if MasterUserPassword is specified.

### Expected behavior

The linter should recognize when `ManageMasterUserPassword` is `true`, and should not report the above error.

### Reproduction template

```yaml
  Cluster:
    Type: AWS::RDS::DBCluster
    Properties:
      [...]
      ManageMasterUserPassword: true
      MasterUsername: example

```

**Comment by PatMyron** [comment]
needs part of these to be removed:
https://github.com/aws-cloudformation/cfn-lint/blob/e26f8036883b52a13aa410f386fa5408212f9613/src/cfnlint/data/AdditionalSpecs/Inclusive.json#L32-L47
